src/main/java/springsamurais/toyswapbackend/service/Comments/CommentsServiceImplementation.java

```diff
@@ -0,0 +1,78 @@
+package springsamurais.toyswapbackend.service.Comments;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import springsamurais.toyswapbackend.exception.CommentFailedToSaveException;
+import springsamurais.toyswapbackend.exception.CommentNotFoundException;
+import springsamurais.toyswapbackend.exception.ListingNotFoundException;
+import springsamurais.toyswapbackend.model.Comment;
+import springsamurais.toyswapbackend.model.Listing;
+import springsamurais.toyswapbackend.repository.CommentsRepository;
+
+import java.util.Date;
+import java.util.List;
+
+@Service
+public class CommentsServiceImplementation implements CommentsService {
+
+    @Autowired
+    private CommentsRepository commentsRepository;
+
+    @Override
+    public List<Comment> getCommentsByUserId(Long userId) {
+        List<Comment> comments = commentsRepository.findByCommenterId(userId);
+        if (comments.isEmpty()) {
+            throw new CommentNotFoundException("No comments found with user ID " + userId);
+        }
+        return comments;
+    }
+
+    @Override
+    public List<Comment> getCommentsByListingId(Long listingId) {
+        List<Comment> comments = commentsRepository.findByListingId(listingId);
+        if (comments.isEmpty()) {
+            throw new CommentNotFoundException("No comments found with listing ID " + listingId);
+        }
+        return comments;
+    }
+
+    @Override
+    public Comment saveComment(Comment comment) {
+        try {
+            comment.setDateCommented(new Date());
```

Comment:
I've noticed dates coming back on **listings** with timestamps of "00:00:00" - is this to do with how the date is saved? This will need amending for comments if it's the same as comments will need a time stamp too.